SNIPS/snip-16.md

```diff
@@ -0,0 +1,46 @@
+# Draft---
+snip: 16
+title: Deprecation of transaction versions 0,1,2
+author: Ilia Volokh <iliav@starkware.co>
+discussions-to: 
+status: Draft
+type: Core
+created: July 2, 2024
+---
+
+## Simple Summary
+In the context of a fee market, transaction versions 0,1,2 are inferior to and incompatible with transaction version 3: they support a different fee token and have inefrior bid structure.
+
+We propose to depracate transaction versions 0,1,2 and to stop supporting them once the mempool is implemented in the sequencer.
```

Comment:
```suggestion
We propose to deprecate transaction versions 0, 1 and 2 and to stop supporting them once the mempool is implemented in the sequencer.
```

Reply:
Could you any timeline on when this mempool should be there?
This allow to better estimate when this change will/should happen